src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs

```diff
@@ -126,15 +138,36 @@ public static IResourceBuilder<AzureStorageResource> RunAsEmulator(this IResourc
         builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(builder.Resource, async (@event, ct) =>
         {
             var connectionString = await builder.Resource.GetBlobConnectionString().GetValueAsync(ct).ConfigureAwait(false);
-
-            if (connectionString == null)
+            if (connectionString is null)
             {
-                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{builder.Resource.Name}' resource but the connection string was null.");
+                throw new DistributedApplicationException($"BeforeResourceStartedEvent was published for the '{builder.Resource.Name}' resource but the connection string was null.");
             }
 
             blobServiceClient = CreateBlobServiceClient(connectionString);
         });
 
+        builder.ApplicationBuilder.Eventing.Subscribe<ResourceReadyEvent>(builder.Resource, async (@event, ct) =>
+        {
+            if (blobServiceClient is null)
+            {
+                throw new DistributedApplicationException($"BlobServiceClient was not created for the '{builder.Resource.Name}' resource.");
+            }
+
+            var connectionString = await builder.Resource.GetBlobConnectionString().GetValueAsync(ct).ConfigureAwait(false);
+            if (connectionString is null)
+            {
+                throw new DistributedApplicationException($"ResourceReadyEvent was published for the '{builder.Resource.Name}' resource but the connection string was null.");
+            }
+
+            if (builder.Resource.BlobsResource is not null)
+            {
+                foreach (var blobContainer in builder.Resource.BlobsResource.BlobContainers)
+                {
+                    await blobServiceClient.GetBlobContainerClient(blobContainer.BlobContainerName).CreateIfNotExistsAsync(cancellationToken: ct).ConfigureAwait(false);
```

Comment:
Should we only do this if it is an emulator? In cases where we deploy bicep, it will exist.

Reply:
I don't think segmenting this is a good idea both from the point of maintainability and the ability to reason about the implementation.
There could be other scenarios where bicep deployment isn't used but and there's an existing storage, but the required container is absent.